packages/components/src/components/trace/TraceFilter.vue

```diff
@@ -0,0 +1,416 @@
+      >
+        <li
+          :class="getSuggestionClasses(index)"
+          v-for="(item, index) in suggestionsList"
+          :key="item"
+          @click.stop="makeSelection(item)"
+        >
+          {{ item }}
+        </li>
+      </ul>
+    </form>
+    <div class="trace-filter__arrows" v-if="nodesLength">
+      <div class="trace-filter__arrow" @click="onPrevArrow">
+        <ArrowSearchLeftIcon />
+      </div>
+      <div class="trace-filter__arrows-text">
+        <span v-if="Number.isInteger(currentIndex)">
+          <b>{{ currentIndex + 1 }}</b>
+          /
+        </span>
+        {{ nodesLength }} results
+      </div>
+      <div class="trace-filter__arrow" @click="onNextArrow">
+        <ArrowSearchRightIcon />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import ArrowSearchLeftIcon from '@/assets/arrow-search-left.svg';
+import ArrowSearchRightIcon from '@/assets/arrow-search-right.svg';
+import CloseThinIcon from '@/assets/close-thin.svg';
+import SearchIcon from '@/assets/search.svg';
+import { getEventTarget } from '@appland/diagrams';
+
+export default {
+  name: 'v-trace-filter',
+
+  components: {
+    ArrowSearchLeftIcon,
+    ArrowSearchRightIcon,
+    CloseThinIcon,
+    SearchIcon,
+  },
+
+  props: {
+    nodesLength: {
+      type: Number,
+      require: false,
+      default: 0,
+    },
+    currentIndex: {
+      type: Number,
+      default: null,
+    },
+    suggestions: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      filterValue: '',
+      showSuggestions: false,
+      selectedSuggestion: false,
+    };
+  },
+
+  watch: {
+    filterValue: {
+      handler(value) {
+        this.$emit('onChange', value);
+      },
+    },
+    selectedSuggestion: {
+      handler(selectedSuggestion) {
+        if (selectedSuggestion === false) {
+          return;
+        }
+
+        const selected =
+          this.$refs.suggestionsList.querySelectorAll('li')[selectedSuggestion];
+
+        if (selected.offsetTop < this.$refs.suggestionsList.scrollTop) {
+          this.$refs.suggestionsList.scrollTop = selected.offsetTop;
+        } else if (
+          selected.offsetTop >=
+          this.$refs.suggestionsList.scrollTop +
+            this.$refs.suggestionsList.offsetHeight -
+            2
+        ) {
+          this.$refs.suggestionsList.scrollTop =
+            selected.offsetTop - selected.offsetHeight * 4;
+        }
+      },
+    },
+  },
+
+  computed: {
+    formClasses() {
+      const classList = ['trace-filter__form'];
+
+      if (this.showSuggestions && this.suggestionsList.length) {
+        classList.push('trace-filter__form--show-suggestions');
+      }
+
+      return classList;
+    },
+    suggestionsList() {
+      let suggestions = this.suggestions
+        .map((e) => e.toString())
+        .filter((e, i, arr) => arr.indexOf(e) === i);
+
+      if (!this.filterValue.endsWith(' ')) {
```

Comment:
This should also work when searching a quoted string, e.g. `"GET /admin"`. The space after `"GET` shouldn't trigger the list to display every item.